src/ServerRequestCreator.php

```diff
@@ -51,13 +51,18 @@ public function fromGlobals(): ServerRequestInterface
 
         $headers = \function_exists('getallheaders') ? getallheaders() : static::getHeadersFromServer($_SERVER);
 
-        return $this->fromArrays($server, $headers, $_COOKIE, $_GET, $_POST, $_FILES, \fopen('php://input', 'r') ?: null);
+        $post = null;
+        if ('POST' === $this->getMethodFromEnv($server) && \in_array($headers['content-type'], ['application/x-www-form-urlencoded', 'multipart/form-data'])) {
```

Comment:
I think that code not case insensitive and not support header values like `application/x-www-form-urlencoded;charset= charset=utf-8`. And it seems not enough `application/json`

Reply:
Are MIME types case insensitive? Maybe they are, I never actually considered. Good catch. Will do some research and patch accordingly. Should definitely support the `;charset` case!

`application/json` is not a POST body that PHP natively parses, so it is not something we can trigger `withParsedBody` for. I would recommend using some sort of middleware (e.g. [Payload](https://github.com/middlewares/payload)) if you need special request parsing.